addons/mail/models/mail_thread.py

```diff
@@ -4531,6 +4531,21 @@ def _message_update_content(self, message, body, attachment_ids=None, partner_id
     # CONTROLLERS
     # ------------------------------------------------------
 
+    def _get_mail_thread_data_access_rights(self, store):
+        """ Return the access rights of the current user on threads in self.
+        The threads are expected to be of the same model.
+        """
+        readable_threads = self._filter_access_rules('read')
+        writable_threads = readable_threads._filter_access_rules('write')
+        has_write_access = self.check_access_rights('write', raise_exception=False)
+        for thread in self:
+            data = {
+                'hasReadAccess': thread in readable_threads,
+                'hasWriteAccess': thread in writable_threads and has_write_access,
+                'canPostOnReadonly': thread._mail_post_access == 'read',
+            }
+            store.add(thread, data, as_thread=True)
```

Comment:
`_thread_to_store` with empty `request_list` (or specifically requesting access rights) should return just that.